### PR TITLE
addMonths & isWithinRange

### PR DESCRIPTION
**Pull request by dmtrKovalenko**

### CODE REVIEW

File: packages/date-fns/src/date-fns-utils.ts [hunk 1]
```diff
@@ -29,7 +29,9 @@ import startOfMonth from "date-fns/startOfMonth";
 import endOfMonth from "date-fns/endOfMonth";
 import startOfWeek from "date-fns/startOfWeek";
 import startOfYear from "date-fns/startOfYear";
+import isWithinRange from "date-fns/isWithinInterval";
```
Comment (oliviertassinari) [inline packages/date-fns/src/date-fns-utils.ts]: Unused import, can be removed

File: packages/date-fns/src/date-fns-utils.ts [hunk 2]
```diff
@@ -235,6 +241,10 @@ export default class DateFnsUtils implements IUtils<Date> {
     return isAfter(date, endOfYear(value));
   }
 
+  public isWithinRange(date: Date, [start, end]: [Date, Date]) {
```
Comment (oliviertassinari) [inline packages/date-fns/src/date-fns-utils.ts]: Did you consider to use `isAfter` and `isBefore` over a new method?

File: packages/dayjs/src/dayjs-utils.ts [hunk 3]
```diff
@@ -158,17 +160,19 @@ export default class DayjsUtils implements IUtils<defaultDayjs.Dayjs> {
...
+    return count < 0 ? date.subtract(Math.abs(count), "month") : date.add(count, "month");
   }
 
   public setMonth(date: Dayjs, count: number) {
-    return date.clone().set("month", count);
+    return date.set("month", count);
   }
 
   public setHours(date: Dayjs, count: number) {
-    return date.clone().set("hour", count);
```
Comment (oliviertassinari) [inline packages/dayjs/src/dayjs-utils.ts]: If dayjs is immutable, maybe we can remove the clone() method form the other wrappers? :)

Comment (dmtrKovalenko) [reply packages/dayjs/src/dayjs-utils.ts]: Yes :)

**Pull request merged**